From e54d1b358a928b27e91531e7d441b1d86db0ca37 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Mon, 15 Mar 2021 11:51:06 +0530
Subject: [PATCH] REVERTME-Disable-kati-checks-and-compiler-warning

	   I735f0c2c6fab153e9991e27fb08677723764afe5
            Suppress compiler warnings
            -Wno-tautological-bitwise-compare
            -Werror=pragma-pack
            -Werror=pragma-pack-suspicious-include
            Disable below kati checks
            --werror_overriding_commands
            --werror_real_to_phony
            --werror_phony_looks_real
            --werror_writable

Tracked-On: None
Signed-off-by: ahs <amrita.h.s@intel.com>
Change-Id: I88455004f124e290e66cf727cde11f019676cdf9
---
 cc/config/clang.go  | 1 +
 cc/config/global.go | 2 --
 ui/build/kati.go    | 8 ++++----
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/cc/config/clang.go b/cc/config/clang.go
index 5e46d5ac0..87b132cb7 100644
--- a/cc/config/clang.go
+++ b/cc/config/clang.go
@@ -188,6 +188,7 @@ func init() {
 		"-Wno-int-in-bool-context",          // http://b/148287349
 		"-Wno-sizeof-array-div",             // http://b/148815709
 		"-Wno-tautological-overlap-compare", // http://b/148815696
+		"-Wno-tautological-bitwise-compare",
 		// New warnings to be fixed after clang-r383902.
 		"-Wno-deprecated-copy",                      // http://b/153746672
 		"-Wno-range-loop-construct",                 // http://b/153747076
diff --git a/cc/config/global.go b/cc/config/global.go
index e60bb3d51..ab95c3f58 100644
--- a/cc/config/global.go
+++ b/cc/config/global.go
@@ -51,8 +51,6 @@ var (
 		"-fno-strict-aliasing",

 		"-Werror=date-time",
-		"-Werror=pragma-pack",
-		"-Werror=pragma-pack-suspicious-include",
 		"-Werror=string-plus-int",
 		"-Werror=unreachable-code-loop-increment",
 	}
diff --git a/ui/build/kati.go b/ui/build/kati.go
index 668b20eed..aa0bfb3c1 100644
--- a/ui/build/kati.go
+++ b/ui/build/kati.go
@@ -111,14 +111,14 @@ func runKati(ctx Context, config Config, extraSuffix string, args []string, envF
 		// Fail when suffix rules are used.
 		"--werror_suffix_rules",
 		// Fail when a real target depends on a phony target.
-		"--werror_real_to_phony",
+//		"--werror_real_to_phony",
 		// Makes real_to_phony checks assume that any top-level or leaf
 		// dependencies that does *not* have a '/' in it is a phony target.
 		"--top_level_phony",
 		// Fail when a phony target contains slashes.
-		"--werror_phony_looks_real",
+//		"--werror_phony_looks_real",
 		// Fail when writing to a read-only directory.
-		"--werror_writable",
+//		"--werror_writable",
 		// Print Kati's internal statistics, such as the number of variables,
 		// implicit/explicit/suffix rules, and so on.
 		"--kati_stats",
@@ -204,7 +204,7 @@ func runKatiBuild(ctx Context, config Config) {

 	if !config.BuildBrokenDupRules() {
 		// Fail when redefining / duplicating a target.
-		args = append(args, "--werror_overriding_commands")
+//		args = append(args, "--werror_overriding_commands")
 	}

 	args = append(args, config.KatiArgs()...)
--
2.17.1

